### ISSUE TITLE
Cleanup the reserved argument for GRPC Core APIs

### ISSUE THREAD

**Issue by yijiem** [grpc/grpc#32554]
This came up in https://github.com/grpc/grpc/pull/32452. The `reserved` argument in some GRPC Core APIs were reserved to provide ABI stability. But it was never used. We need to open a gRFC to delete all `void* reserved` argument from public APIs. This will also need a cherrypick to Google3 with changes of internal usage. See https://github.com/grpc/grpc/pull/32452#issuecomment-1440699975 for details.

**Comment by markdroth** [comment]
I don't think we need to track this separately.  At some point after EventEngine is done, we'll probably redesign the C-core API in C++ anyway, and this issue will go away as part of that work.
